sre_maturity.md

```diff
@@ -175,27 +175,68 @@ In this phase you do ad-hoc security assessments, scans and pentesting.
+4. The team must have the technical and organizational capability to work on toil reduction – at a minimum a repository to store runbooks, guidelines on creating automations and the ability to fix and/or file bugs with the product you're SRE-ing.
+5. Oncall shifts must (on average) be light enough that when toil gets noticed, oncall engineers have sufficient time and energy to tend to it – either address it immediately if it's small enough (e.g. update a runbook) or write it up for the team to follow up on later.
 
-### Room to improve - Phase 2: The 'Walk' Phase
+There are some additional considerations when it comes to developer teams – as opposed to dedicated SRE teams – trying to take on SRE tasks:
+1. The non-oncall time needs to be split between standard dev work (like customer-visible features) and SRE-related dev work.
+   We strongly suggest initially reserving at least half of the total dev work time for SRE-related tasks.
+   Giving a dev team SRE responsibilities without reserving sufficient time for tending to an SRE fundamental – toil reduction – will be setting that team up for failure.
+2. Do not expect your engineers to handle development tasks during oncall shifts.
+   Oncall needs to be reserved for responding to interrupts and following up on those responses.
+   Consistent and thorough followups are a crucial prerequisite for staying ahead of toil growth.
 
-*This is a stub. Please help by adding content. Ref:[SIGSRE-86](https://issues.redhat.com/browse/SIGSRE-86)*
+At this stage it isn't necessary to measure and compare different kinds of work, that will come later.
+Automating away unpleasant toil and enhancing one's oncall experience are self-rewarding tasks.
+If engineers are given the time, know-how and the tools to work on toil reduction, they should be able to stay on top of toil growth by focusing on what they perceive to be the most bothersome issues.
+Still, leadership should regularly reinforce the message that constant improvement to operational efficiency is a key component of SRE success and a lens through which engineers should always look at their work.
 
-* Start measuring Toil more formally & management track it
-  * Develop relationship with the upstream (eng and sre, or sre & upstream product) for managing toil
+#### Toil Bankruptcy and the Dangers of Unmanaged Toil
 
-### Room to improve - Phase 3: The 'Run' Phase
+Should an SRE team fail at reducing toil faster than it accumulates due to service growth, such a team can find itself in a very serious predicament – spending all of its time on operations tasks with zero room to improve (e.g. automate) anything.
 
-*This is a stub. Please help by adding content. Ref:[SIGSRE-87](https://issues.redhat.com/browse/SIGSRE-87)*
+The chart below visualizes what happens to an SRE team's time commitments when their system is growing 20% quarter over quarter yet they do not address the toily part of operations (marked in red).
```

Comment:
Can you please add an attribution + context to where this data came from?

Reply:
I've created this chart with a simple [sheet](https://docs.google.com/spreadsheets/d/1x3THIRbH1cCtlZAi6KMaRV_VW53jbtrBnxlq4B8Uyrs/edit#gid=0), but since it's in our Red Hat sheets, I can't link to that directly.

Hm, what I could do is add a short separate document with the raw data in a table and an explanation of how the calculations were derived and link to that.

Reply:
Added the detailed explanation. (Linked at the end of that paragraph).